lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -676,42 +650,49 @@ WasmBytecodeGenerator::EmitSetLocal(bool tee)
     }
 }
 
-template<WasmTypes::WasmType type>
 EmitInfo
-WasmBytecodeGenerator::EmitConst()
+WasmBytecodeGenerator::EmitConst(WasmTypes::WasmType type, WasmConstLitNode cnst)
 {
-    WasmRegisterSpace * regSpace = GetRegisterSpace(type);
-
-    Js::RegSlot tmpReg = regSpace->AcquireTmpRegister();
+    Js::RegSlot tmpReg = GetRegisterSpace(type)->AcquireTmpRegister();
     EmitInfo dst(tmpReg, type);
-    EmitLoadConst(dst, GetReader()->m_currentNode.cnst);
+    EmitLoadConst(dst, cnst);
     return dst;
 }
 
-void WasmBytecodeGenerator::EmitLoadConst(EmitInfo dst, WasmConstLitNode cnst)
+void
+WasmBytecodeGenerator::EmitLoadConst(EmitInfo dst, WasmConstLitNode cnst)
 {
     switch (dst.type)
     {
     case WasmTypes::F32:
-        m_writer.AsmFloat1Const1(Js::OpCodeAsmJs::Ld_FltConst, dst.location, cnst.f32);
+        m_writer->AsmFloat1Const1(Js::OpCodeAsmJs::Ld_FltConst, dst.location, cnst.f32);
         break;
     case WasmTypes::F64:
-        m_writer.AsmDouble1Const1(Js::OpCodeAsmJs::Ld_DbConst, dst.location, cnst.f64);
+        m_writer->AsmDouble1Const1(Js::OpCodeAsmJs::Ld_DbConst, dst.location, cnst.f64);
         break;
     case WasmTypes::I32:
-        m_writer.AsmInt1Const1(Js::OpCodeAsmJs::Ld_IntConst, dst.location, cnst.i32);
+        m_writer->AsmInt1Const1(Js::OpCodeAsmJs::Ld_IntConst, dst.location, cnst.i32);
         break;
     case WasmTypes::I64:
-        m_writer.AsmLong1Const1(Js::OpCodeAsmJs::Ld_LongConst, dst.location, cnst.i64);
+        m_writer->AsmLong1Const1(Js::OpCodeAsmJs::Ld_LongConst, dst.location, cnst.i64);
         break;
     default:
         throw WasmCompilationException(_u("Unknown type %u"), dst.type);
     }
 }
 
+WasmConstLitNode
+WasmBytecodeGenerator::GetZeroCnst()
+{
+    WasmConstLitNode cnst;
```

Comment:
pretty sure you don't need memset and can just do `= {0};`, though maybe the compiler can figure that out too and avoid the call